contracts/Manager.sol

```diff
@@ -722,97 +429,179 @@ contract Manager is UUPSOwnableUpgradeable, UsingRegistryUpgradeable {
+     * @notice Checks if strategy was deprecated. Deprecated strategy is reverted to default.
+     * Updates the strategies.
+     * @param accountAddress The account.
+     * @param strategy The strategy.
+     * @return Up to date strategy
      */
-    function isGroupMemberElected(address groupMember) private view returns (bool) {
-        IElection election = getElection();
-
-        address[] memory electedValidatorSigners = election.electValidatorSigners();
+    function _checkAndUpdateStrategy(address accountAddress, address strategy)
+        private
+        returns (address)
+    {
+        address checkedStrategy = _checkStrategy(strategy);
+        if (checkedStrategy != strategy) {
+            strategies[accountAddress] = checkedStrategy;
+        }
+        return checkedStrategy;
+    }
 
-        for (uint256 i = 0; i < electedValidatorSigners.length; i++) {
-            if (electedValidatorSigners[i] == groupMember) {
-                return true;
-            }
+    /**
+     * @notice Schedules transfer of Celo between strategies.
+     * @param fromStrategy The from validator group.
+     * @param toStrategy The to validator group.
+     * @param stCeloAmount The stCelo amount.
+     * @param from The from address.
+     * @param to The to address.
+     */
+    function _transfer(
+        address fromStrategy,
+        address toStrategy,
+        uint256 stCeloAmount,
+        address from,
+        address to
+    ) private {
+        fromStrategy = _checkAndUpdateStrategy(from, fromStrategy);
+        toStrategy = _checkAndUpdateStrategy(to, toStrategy);
+
+        if (fromStrategy == toStrategy) {
+            // either both addresses use default strategy
+            // or both addresses use same allowed strategy
+            return;
         }
 
-        return false;
+        _transferWithoutChecks(fromStrategy, toStrategy, stCeloAmount);
     }
 
     /**
-     * @notice Returns the storage, major, minor, and patch version of the contract.
-     * @return Storage version of the contract.
-     * @return Major version of the contract.
-     * @return Minor version of the contract.
-     * @return Patch version of the contract.
-     */
-    function getVersionNumber()
-        external
-        pure
-        returns (
-            uint256,
-            uint256,
-            uint256,
-            uint256
-        )
-    {
-        return (1, 2, 0, 1);
+     * @notice Schedules transfer of Celo between strategies.
+     * @param fromStrategy The from validator group.
+     * @param toStrategy The to validator group.
+     * @param stCeloAmount The stCelo amount.
+     */
+    function _transferWithoutChecks(
+        address fromStrategy,
+        address toStrategy,
+        uint256 stCeloAmount
+    ) private {
+        address[] memory fromGroups;
+        uint256[] memory fromVotes;
+        (fromGroups, fromVotes) = distributeVotes(toCelo(stCeloAmount), fromStrategy);
```

Comment:
Shouldn't this be `distributeWithdrawals`?

Reply:
No, since we are not withdrawing but rather revoking